packages/react-native/ReactCommon/react/renderer/components/textinput/platform/ios/react/renderer/components/iostextinput/TextInputEventEmitter.cpp

```diff
@@ -188,4 +188,30 @@ void TextInputEventEmitter::dispatchTextInputContentSizeChangeEvent(
   });
 }
 
+#if TARGET_OS_OSX // [macOS
+void TextInputEventEmitter::onAutoCorrectChange(OnAutoCorrectChange event) const {
+  dispatchEvent("autoCorrectChange", [event=std::move(event)](jsi::Runtime &runtime) {
```

Comment:
This struct consists only of PODs. It's faster to just copy the whole thing. But we're only reading a boolean from this struct. Can we just copy the value instead?

Reply:
Looking at the rest of this file, looks like we don't use `std::move` elsewhere. I'll update this to match the rest

Reply:
It looks like we should also refactor to use `textInputMetrics` insteaad of custom events

Reply:
Resolved this, but the refactor opens a few more questions. I'll add a few comments outside of this thread.